dev/kubecdl/main.go

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+)
+
+var (
+	project    = pflag.StringP("project", "p", "", "name of the Google project - defaults to what's configured in gcloud")
+	kubeconfig = pflag.StringP("kubeconfig", "k", os.Getenv("KUBECONFIG"), "kubeconfig filepath")
+)
+
+func main() {
+	pflag.Parse()
+
+	clusterName := pflag.Arg(0)
+	if clusterName == "" {
+		logrus.Fatalf("usage: %s [--project|-p GoogleProject] [--kubeconfig|-k ~/.kube/config] <clusterName>", os.Args[0])
+	}
+
+	_, err := exec.LookPath("gcloud")
+	if err != nil {
+		logrus.WithError(err).Fatal("gcloud is not available")
+	}
+
+	prj := *project
+	if prj == "" {
+		out, err := exec.Command("gcloud", "config", "get-value", "project").CombinedOutput()
+		if err != nil {
+			logrus.WithError(err).Fatal("cannot get configured project. Use --project to explicitly set one.")
+		}
+		prj = strings.TrimSpace(string(out))
+	}
+	kubecfgfn := *kubeconfig
+	if kubecfgfn == "" {
+		home, err := os.UserHomeDir()
+		if err != nil {
+			logrus.Fatal(err)
+		}
+		kubecfgfn = filepath.Join(home, ".kube", "config")
+	}
+
+	serverIP, err := getServerIP(clusterName, prj)
```

Comment:
you can use the external LB `gcloud compute forwarding-rules  describe server-ws-<cluster name>-external`